lib/Dialect/Arc/Transforms/ArcCanonicalizer.cpp

```diff
@@ -709,15 +708,14 @@ LogicalResult KeepOneVecOp::matchAndRewrite(VectorizeOp vecOp,
   for (auto [argIdx, inputVec] : llvm::enumerate(vecOp.getInputs())) {
     auto input = SmallVector<Value>(inputVec.begin(), inputVec.end());
     if (auto in = inExists.find(input); in != inExists.end()) {
-      argsToRemove.set(argIdx);
       rewriter.replaceAllUsesWith(currentBlock.getArgument(argIdx - shuffledBy),
                                   currentBlock.getArgument(in->second));
       currentBlock.eraseArgument(argIdx - shuffledBy);
       ++shuffledBy;
       changed = true;
       continue;
     }
-    inExists[input] = argIdx;
+    inExists[input] = argIdx - shuffledBy;
```

Comment:
Nit: instead of incrementing `argIdx` every iteration and subtracting the number of args we have already deleted, we could only increment `argIdx` when we don't delete the argument.

Reply:
Yes, that's better. can we handle this in `auto [argIdx, inputVec] : llvm::enumerate(vecOp.getInputs())`, should I refactor the loop?